lib/game.rb

```diff
@@ -17,4 +17,45 @@ def lowest_total_score
     end.min
   end
   
+  def average_goals_per_game
+    goals = []
+    @game_data.each do |game|
+      goals_sum = game[:away_goals].to_i + game[:home_goals].to_i
+      goals << goals_sum
+    end
+    goals.sum / goals.length.to_f
+  end
+
+  def count_of_goals_by_season
+    goals_by_season = {}
+    @game_data.each do |game|
+      if goals_by_season[game[:season]].nil?
+        goals_by_season[game[:season]] = game[:home_goals].to_i + game[:away_goals].to_i
+      else
+        goals_by_season[game[:season]] += game[:home_goals].to_i + game[:away_goals].to_i
+      end
+    end
+    goals_by_season
+  end
+
+  def count_of_games_by_season
+    games_by_season = {}
+    @game_data.each do |game|
+      if games_by_season[game[:season]].nil?
+        games_by_season[game[:season]] = 1
+      else
+        games_by_season[game[:season]] += 1
+      end
+    end
+    games_by_season
+  end
+
+  # def average_goals_by_season
```

Comment:
average_goals_by_season is currently not working (we worked on this as a group) b/c count_of_games_by_season is returning an empty hash (cant divide by nil) we plan on merging this and fixing it later